datanode/entities/data_source_definition.go

```diff
@@ -0,0 +1,216 @@
+package entities
+
+import (
+	"code.vegaprotocol.io/vega/core/types"
+	"code.vegaprotocol.io/vega/protos/vega"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+type DataSourceDefinition struct {
+	*vega.DataSourceDefinition
+}
+
+func (s DataSourceDefinition) MarshalJSON() ([]byte, error) {
+	return protojson.Marshal(s)
+}
+
+func (s *DataSourceDefinition) UnmarshalJSON(b []byte) error {
+	s.DataSourceDefinition = &vega.DataSourceDefinition{}
+	return protojson.Unmarshal(b, s)
+}
+
+func (s *DataSourceDefinition) GetOracle() (*DataSourceSpecConfiguration, error) {
+	ds := &DataSourceSpecConfiguration{
+		Signers: Signers{},
+		Filters: []Filter{},
+	}
+
+	if s.DataSourceDefinition != nil {
+		if s.SourceType != nil {
+			switch dst := s.SourceType.(type) {
+			case *vega.DataSourceDefinition_External:
+				if dst.External != nil {
+					if dst.External.SourceType != nil {
+						switch dste := dst.External.SourceType.(type) {
```

Comment:
Same here, this many levels should be possible to avoid here

Reply:
Right, I'll fix this everywhere, thanks!